pennylane/estimator/resources_base.py

```diff
@@ -0,0 +1,407 @@
+        """Add two resources objects in series"""
+        assert isinstance(other, self.__class__)
+        return add_in_series(self, other)
+
+    def __and__(self, other: Resources) -> Resources:
+        """Add two resources objects in parallel"""
+        assert isinstance(other, self.__class__)
+        return add_in_parallel(self, other)
+
+    def __eq__(self, other: Resources) -> bool:
+        """Determine if two resources objects are equal"""
+        return (self.gate_types == other.gate_types) and (self.wire_manager == other.wire_manager)
+
+    def __mul__(self, scalar: int) -> Resources:
+        """Scale a resources object in series"""
+        assert isinstance(scalar, int)
+        return mul_in_series(self, scalar)
+
+    def __matmul__(self, scalar: int) -> Resources:
+        """Scale a resources object in parallel"""
+        assert isinstance(scalar, int)
+        return mul_in_parallel(self, scalar)
+
+    __rmul__ = __mul__
+    __radd__ = __add__
+    __rand__ = __and__
+    __rmatmul__ = __matmul__
+
+    @property
+    def clean_gate_counts(self):
+        r"""Produce a dictionary which stores the gate counts
+        using the operator names as keys.
+
+        Returns:
+            dict: A dictionary with operator names (str) as keys
+                and the number of occurances in the circuit (int) as values.
+        """
+        clean_gate_counts = defaultdict(int)
+
+        for cmp_res_op, counts in self.gate_types.items():
+            clean_gate_counts[cmp_res_op.name] += counts
+
+        return clean_gate_counts
+
+    def __str__(self):
+        """Generates a string representation of the Resources object."""
+        wm = self.wire_manager
+        total_qubits = wm.total_qubits
+        total_gates = sum(self.clean_gate_counts.values())
+
+        total_gates_str = str(total_gates) if total_gates <= 999 else f"{Decimal(total_gates):.3E}"
+        total_qubits_str = (
+            str(total_qubits) if total_qubits <= 9999 else f"{Decimal(total_qubits):.3E}"
+        )
+
+        items = "--- Resources: ---\n"
+        items += f" Total qubits: {total_qubits_str}\n"
+
+        qubit_breakdown_str = f"    algorithmic qubits: {wm.algo_qubits}\n    allocated qubits: {wm.clean_qubits+wm.dirty_qubits}\n\t clean qubits: {wm.clean_qubits}\n\t dirty qubits: {wm.dirty_qubits}\n"
+        items += qubit_breakdown_str
+
+        items += f" Total gates : {total_gates_str}\n  "
+
+        gate_type_str = ",\n  ".join(
+            [
+                f"'{gate_name}': {Decimal(count):.3E}" if count > 999 else f"'{gate_name}': {count}"
+                for gate_name, count in self.clean_gate_counts.items()
+            ]
+        )
+        items += gate_type_str
+
+        return items
+
+    def __repr__(self):
+        """Compact string representation of the Resources object"""
+        return {
+            "wire_manager": self.wire_manager,
+            "gate_types": self.gate_types,
+        }.__repr__()
```

Comment:
Can we make it clear we are dealing with a `Resources` class here and not a dictionary? I could see that causing a lot of confusion if people think they are seeing a dictionary when it's actually a `Resource` object.